### DISCUSSION TITLE
Client Secret changed - No Access possible

### DISCUSSION THREAD

**Discussion by Lemodo20** [abuzuhri/Amazon-SP-API-CSharp#758] [General]
Hi,

i've got an email from Amazon, that my Client Secret have to be renewed in a few days. So i renewd it and was already wondering, that the secret looks now different. Before it looks like this:

f05c0f3b44f40c526bxxxxxxexxxacxxxxcaexxxebxxxxxxxxxxxxxxxxxxxxx
and now like this:
amzn1.oa2-cs.v1.923fb9616ddxacxxxxcaexxxebxxxxxxxxxxxxxxxxxxxxx

So special the front part was not existing before. With this new Secret i can't access any more. If i use the old one (this is still valid for 7 days) all works ok. So what did i made wrong?

**Comment by zinsmeik** [comment]
Hi,
when I changed our Client Secrect I also needed to generate a new Refreshtoken to make it work.
However the format of the Client Secret was the same.

**Comment by Lemodo20** [comment]
Based on the help document, there is no new refresh token needen:
https://developer-docs.amazon.com/sp-api/docs/rotating-your-apps-lwa-credentials

Do you need to generate new refresh tokens when you rotate LWA credentials?

No. Refresh tokens are tied to the LWA client identifier. Upon LWA rotation, a new LWA client secret will be created. You will need to use this new secret along with its existing client identifier and refresh token to retrieve new access tokens. Refresh tokens do not need to be recreated.

**Reply by Lemodo20** [reply]
I changed now also the refresh token... Error is still "Error getting LWA Access token"

**Comment by Lemodo20** [comment]
I received this answer from Amazon Developer Support:

They have informed us that the new Client Secret is 81 characters long, while the existing one is 65 characters long. 

Please update your code to accept the new longer string. The newly generated Client Secret will be prefixed with "amzn1.oa2 -cs.v1." followed by an alphanumeric string. Without the prefix, your calls to SPAPI will fail. 

There are no measures that can be taken from our side. Thank you for your understanding.

So this is not really helpful. I use this API since long time and where should i change the lenght of the secret? (I use VB.net to access it). Why should be lenght be a problem?

**Comment by Lemodo20** [comment]
I've downloaded me now the C# Project from Github to debug it there (but don't have good C# knowledge. So looks like the code internal handle the clientsecret as string and it have the correnct lenght/content when going through the debugger. But this function call fails:

var response = await RestClient.ExecuteAsync(accessTokenRequest, cancellationToken).ConfigureAwait(false);
with this exception "Error getting LWA Access Token" 

All client variable are set correct and shown in the debugger. If i replace the client Secret with my old one, all works at this point. So why can the client secret be wrong? Based on Amazon documentation the OLD and NEW should be working at the same time (for 7 days)

**Comment by Lemodo20** [comment]
Problem solved. The problem was in front of the PC. Stupid dialog from Amazon show only a part of the secret. When you mark it by the mouse, it don't scroll to the right. Only if you double click the text, it's fully market. So my client secret was always to short. I copied it one time and works always with same wrong data ;-(
